flink-table/flink-table-common/src/main/java/org/apache/flink/table/catalog/CatalogTable.java

```diff
@@ -29,4 +32,15 @@
 	 * @return table statistics
 	 */
 	TableStats getStatistics();
+
+	/**
+	 * Check if the table is partitioend or not.
+	 */
+	boolean isPartitioned();
+
+	/**
+	 * Get the partition keys of the table. This will be an empty set if the table is not partitioned.
+	 * @return partition keys of the table.
+	 */
+	LinkedHashSet<String> getPartitionKeys() throws TableNotPartitionedException;
```

Comment:
I find that Hive doesn't allow partitioned keys in table schema, do we have this restriction?

Reply:
I don't really understand your question. Can you elaborate? 

You are right that Hive's [Table](https://hive.apache.org/javadocs/r2.3.4/api/org/apache/hadoop/hive/metastore/api/Table.html) uses a separate field dedicated for partition keys.